corehq/apps/locations/models.py

```diff
@@ -353,7 +353,7 @@ class SQLLocation(MPTTModel):
     metadata = jsonfield.JSONField(default=dict, blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
     last_modified = models.DateTimeField(auto_now=True)
-    is_archived = models.BooleanField(default=False)
+    is_archived = models.BooleanField(default=False, db_index=True)
```

Comment:
yeah i saw that query you had pasted into slack. the biggest factor impacting performance on that is the enormous IN() clause - if we rewrite that to use a temp table with the case ids and join to it, it would perform much better.

i think even just adding domain to the WHERE clause might make a difference there too, since that would cut the result set down by a lot so we don't have to check the IN on the stockstate for practically all locations